python/ray/serve/_private/proxy_state.py

```diff
@@ -30,35 +33,282 @@
 logger = logging.getLogger(SERVE_LOGGER_NAME)
 
 
+class ActorWrapperCallStatus(str, Enum):
+    PENDING = "PENDING"
+    FINISHED_SUCCEED = "FINISHED_SUCCEED"
+    FINISHED_FAILED = "FINISHED_FAILED"
+
+
+class ActorWrapper(ABC):
+    @property
+    @abstractmethod
+    def actor_id(self) -> str:
+        """Return the actor id of the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def start_new_ready_check(self):
+        """Start a new ready check on the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def start_new_health_check(self):
+        """Start a new health check on the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def start_new_drained_check(self):
+        """Start a new drained check on the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_ready(self) -> Any:
+        """Return the payload from proxy ready check when ready."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_healthy(self) -> bool:
+        """Return whether the proxy actor is healthy or not."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_drained(self) -> bool:
+        """Return whether the proxy actor is drained or not."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_shutdown(self):
+        """Return whether the proxy actor is shutdown."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def update_draining(self, draining: bool):
+        """Update the draining status of the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def kill(self):
+        """Kill the proxy actor."""
+        raise NotImplementedError
+
+
+class ProxyActorWrapper(ActorWrapper):
```

Comment:
The naming here is backwards :)

What you current have defined as `ActorWrapper` is actually a `ProxyWrapper` -- it's an interface defining how to interact w/ a proxy. There is no notion of a "actor" exposed at all.

The implementation is an `ActorProxyWrapper` -- a type of `ProxyWrapper` that interacts with a proxy that happens to be a Ray actor.